tests/test_meshtastic_utils_connect.py

```
@@ -626,3 +784,1023 @@ def test_none_startup_drain_event_is_safe_noop(reset_meshtastic_globals):
+            {"meshtastic": {}},
+            {"meshtastic": {"connection_type": None}},
+        ]
+
+        for config in invalid_configs:
+            with self.subTest(config=config):
+                with patch("mmrelay.meshtastic_utils.logger"):
+                    result = connect_meshtastic(config)
+                    self.assertIsNone(result)
+
+    def test_timeout_breaks_on_shutdown(self):
+        def _timeout_then_shutdown(*_args: Any, **_kwargs: Any) -> NoReturn:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*_args`

(ANN401)